File: 0875_Koko Eating Bananas/KokoEatingBananas.py
class Solution(object):
    def minEatingSpeed(self, piles, H):
        """
        :type piles: List[int]
        :type H: int
        :rtype: int
        """
        l = 1
        r = max(piles)+1
        while l < r:
            m = l + (r - l)//2
            h = 0
            for p in piles:
                h += p/m
                if p % m != 0:
                    h += 1
            if h <= H:
                r = m
            else:
                l = m + 1
        return l

File: 0875_Koko Eating Bananas/KokoEatingBananas.cs
public class Solution {
    public int MinEatingSpeed(int[] piles, int h) {
        var l = 1;
        var r = piles.Max() + 1;
        
        while(l < r)
        {
            var m = (r-l)/2 + l;
            var h2 = 0;
            foreach(var p in piles)
            {
                h2 += p/m;
                if(p % m !=0)
                    h2 += 1;
            }
            
            if(h2 <= h){
                r = m;
            }else{
                l = m + 1;
            }
        }
        
        return l;
    }
}